pom.xml

```diff
@@ -64,6 +65,11 @@
 			<type>jar</type>
 			<scope>compile</scope>
 		</dependency>
+		<dependency>
+			<groupId>com.kohlschutter.junixsocket</groupId>
+			<artifactId>junixsocket-native-common</artifactId>
```

Comment:
I don't like the idea of adding one more dependency* just to support one feature.

\* compile time

Reply:
Yeah, I wasn't too mad about it as well, but thought it would be fine since it's a small dependency. So I think we have a few options (ordered by most to least favourite by me, but open to suggestions):
* [My preferred suggestion] Make the dependency an optional one and do a reflection check to see if it is and if the feature can be supported
* Shade and include the dependency (also since it's tiny), this way it would be self included and no need for consumers to worry
* Accept the dependency as is now since it's tiny
* Not include it at all and follow your below comments passing a socket straight in the constructor. It's my least preferred because I think the consumer shouldn't need to worry about creating/configuring the socket, specially that Jedis has some good defaults there. In the below comment you mention *other* sockets, any specific ones in mind?

(Sorry for the delayed response, this escaped me somehow...)